docs/high-performance/MemoryOwner.md

```diff
@@ -0,0 +1,72 @@
+---
+title: MemoryOwner&lt;T>
+author: Sergio0694
+description: A buffer type implementing `IMemoryOwner<T>` that rents memory from a shared pool
+keywords: windows 10, uwp, windows community toolkit, uwp community toolkit, uwp toolkit, parallel, high performance, net core, net standard
+dev_langs:
+  - csharp
+---
+
+# MemoryOwner&lt;T>
+
+The [MemoryOwner&lt;T>](https://docs.microsoft.com/dotnet/api/microsoft.toolkit.highperformance.buffers.memoryowner-1) is a buffer type implementing `IMemoryOwner<T>`, an embedded length property and a series of performance oriented APIs. It is essentially a lightweight wrapper around the `ArrayPool<T>` type, with some additional helper utilities.
+
+## How it works
+
+`MemoryOwner<T>` has the following main features:
+
+- One of the main issues of arrays returned by the `ArrayPool<T>` APIs and of the `IMemoryOwner<T>` instances returned by the `MemoryPool<T>` APIs is that the size specified by the user is only being used as a _minum_ size: the actual size of the returned buffers might actually be greater. `MemoryOwner<T>` solves this by also storing the original requested size, so that `Memory<T>` and `Span<T>` instances retrieved from it will never need to be manually sliced.
```

Comment:
Could be handy for each of these call outs to the BCL to have examples showing 'if you did this before, do this now...' and get XYZ benefit...

Reply:
Improved the sample for `MemoryOwner<T>` in https://github.com/MicrosoftDocs/WindowsCommunityToolkitDocs/pull/301/commits/8a0d6b8f2147cdb778f08e1a79916dc0023c3be6, let me know what you think! 😊